src/stdio.rs

```diff
@@ -0,0 +1,243 @@
+use super::Error;
+
+use core::mem::ManuallyDrop;
+
+use core::pin::Pin;
+use core::task::{Context, Poll};
+
+use std::fs::File;
+use std::io::{self, IoSlice};
+use std::os::unix::io::{AsRawFd, FromRawFd, IntoRawFd, RawFd};
+
+use std::process;
+
+use tokio::io::{AsyncRead, AsyncReadExt, AsyncWrite, ReadBuf};
+
+#[derive(Debug)]
+pub(crate) enum StdioImpl {
+    /// Read/Write to /dev/null
+    Null,
+    /// Read/Write to a newly created pipe
+    Pipe,
+    /// Read/Write to custom fd
+    Fd(File),
+}
+
+/// Similar to std::process::Stdio
+#[derive(Debug)]
+pub struct Stdio(pub(crate) StdioImpl);
+impl Stdio {
+    /// Create a pipe for child communication
+    pub const fn piped() -> Self {
+        Self(StdioImpl::Pipe)
+    }
+
+    /// Pass /dev/null for child communication
```

Comment:
We should use the exact same documentation for each of these methods as `std::process::Stdio`.

Reply:
Doc updated.